polaris-react/src/components/MediaQueryProvider/MediaQueryProvider.tsx

```diff
@@ -13,9 +13,7 @@ interface Props {
 export const MediaQueryProvider = function MediaQueryProvider({
   children,
 }: Props) {
-  const [isNavigationCollapsed, setIsNavigationCollapsed] = useState(
-    navigationBarCollapsed().matches,
```

Comment:
Do we not need a `useLayoutEffect` or something similar to update this state on render?

Reply:
I think we have some util that wraps `useLayoutEffect` which could be tested here but I think it's fine the way it is